### PR TITLE
DOC-711: Added documentation and legal notices for the new emoticon image database

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: DOC-711

Description of Changes:
* Adds details about the new `emoticons_database` and `emoticons_images_url` settings.
* Adds details and legal notices about the new `emojiimages` database.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: plugins/emoticons.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ tinymce.init({
 
 ## Browser emoji support
 
-The emoticon plugin inserts unicode character codes, such as `\ud83d\ude03` for the smiley emoji. How emoji are rendered is dependent on the web browser and operating system of the user. As a result of this, some emoji may be rendered in black and white, or may not render.
+The emoticon plugin inserts unicode character codes by default, such as `\ud83d\ude03` for the smiley emoji. How emoji are rendered is dependent on the web browser and operating system of the user. As a result of this, some emoji may be rendered in black and white, or may not render.
```
Comment (lnewson) [inline plugins/emoticons.md:33]: I debated about changing this, but didn't want to advertise the new image database too much. As such I just mentioned this applies to the plugin by default as this is what most users will run into.

File: plugins/emoticons.md [hunk 2]
```diff
@@ -107,6 +133,25 @@ tinymce.init({
 });
 ```
 
+### `emoticons_images_url`
+
+This option provides the base URL for the images used to represent emojis when using the `emojiimages` database.
```
Comment (lnewson) [inline plugins/emoticons.md:138]: Do we want to document where/how to get the twemoji graphics to self-host them?

File: plugins/emoticons.md [hunk 3]
```diff
@@ -78,9 +78,35 @@ tinymce.init({
 });
 ```
 
+### `emoticons_database`
+
+This option provides the ability to specify which built-in emoji database to use when rendering emojis in the editor. The following built-in emoji databases are available:
+* `emojis` - This database uses unicode characters to represent emoji in the editor content.
```
Comment (tylerkelly13) [inline plugins/emoticons.md:84]:
```suggestion
* `emojis` - This database uses Unicode characters to represent emoji in the editor content.
```

File: plugins/emoticons.md [hunk 4]
```diff
@@ -78,9 +78,35 @@ tinymce.init({
...
+* `emojiimages` - This database uses images provided by Twitter Emoji (twemoji) to represent emoji in the editor content.
```
Comment (tylerkelly13) [inline plugins/emoticons.md:85]:
```suggestion
* `emojiimages` - This database uses images provided by the Twitter Emoji (twemoji) library to represent emoji in the editor content.
```

File: plugins/emoticons.md (hunk 2 above)
Comment (tylerkelly13) [reply plugins/emoticons.md:138]: yes please.

Comment (tylerkelly13) [reply plugins/emoticons.md:138]: Also, maybe:
```suggestion
This option sets the base URL for the images used to represent emojis when using the `emojiimages` database.
```

**Review by tylerkelly13 — APPROVED** [review]

File: plugins/emoticons.md (hunk 4 above)
Comment (lnewson) [reply plugins/emoticons.md:85]: Hmm... it's not really the library we're using. Maybe `graphics` instead?

Comment (tylerkelly13) [reply plugins/emoticons.md:85]: hmmm project?

Comment (lnewson) [reply plugins/emoticons.md:85]: That works :+1:

**Review by tylerkelly13 — APPROVED** [review]
LGTM :shipit:

**Review by ltrouton — APPROVED** [review]

**Review by EkimChau — APPROVED** [review]

**Pull request merged**